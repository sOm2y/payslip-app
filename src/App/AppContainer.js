/**
 * 
 * the app container, should be take the whole page.
 * using flex box layout
 * with approprate background art
 * 
 */
