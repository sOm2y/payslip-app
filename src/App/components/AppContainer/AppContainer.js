//@flow

/**
 * 
 * the app container, should be take the whole page.
 * using flex box layout
 * with approprate background art
 * 
 */

import React from 'react';
import type { Node } from 'react';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
type Props = {
  children: Node,
  classes: {
    container: string,
    ribbon: string,
    childrenWrapper: string,
    headlineWrapper: string
  }
};

const styles = (theme) => ({
  container: {
    height: '100%',
    width: '100%',
    display: 'flex'
  },
  ribbon: {
    height: 200,
    width: '100%',
    position: 'fixed',
    zIndex: -1,
    backgroundColor: theme.palette.primary[500]
  },
  childrenWrapper: {
    paddingTop: 48,
    paddingLeft: 32
  },
  headlineWrapper: {
    padding: [12, 32],
    color: theme.palette.common.white
  }
});

class AppContainer extends React.PureComponent<Props> {
  render() {
    const { children, classes } = this.props;

    return (
      <div className={classes.container}>
        <div className={classes.ribbon}>
          <div className={classes.headlineWrapper}>
            <Typography type="headline" color="inherit">
              The Payslip App
            </Typography>
          </div>
        </div>
        <div className={classes.childrenWrapper}>{children}</div>
      </div>
    );
  }
}

export default withStyles(styles)(AppContainer);
