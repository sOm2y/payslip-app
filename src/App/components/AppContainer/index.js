//@flow

export { default } from './AppContainer';
