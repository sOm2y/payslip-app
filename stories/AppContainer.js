//@flow

import React from 'react';

import Paper from 'material-ui/Paper';

import AppContainer from '../src/App/components/AppContainer';

export default () => (
  <AppContainer>
    <Paper
      style={{
        height: 500,
        width: 400,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
      }}
    >
      a huge paper
    </Paper>
  </AppContainer>
);
