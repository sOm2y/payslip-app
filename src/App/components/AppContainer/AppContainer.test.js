//@flow
import React from 'react';
import AppContainer from './AppContainer';
import renderer from 'react-test-renderer';

describe('AppContainer', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<AppContainer>hello</AppContainer>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
