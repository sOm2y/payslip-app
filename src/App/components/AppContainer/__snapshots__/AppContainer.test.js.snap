// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer should renders correctly 1`] = `
<div
  className="AppContainer-container-1"
>
  <div
    className="AppContainer-ribbon-2"
  >
    <div
      className="AppContainer-headlineWrapper-4"
    >
      <h1
        className="MuiTypography-root-5 MuiTypography-headline-10 MuiTypography-colorInherit-24"
      >
        The Payslip App
      </h1>
    </div>
  </div>
  <div
    className="AppContainer-childrenWrapper-3"
  >
    hello
  </div>
</div>
`;
